lxd/daemon.go

```diff
@@ -669,6 +674,88 @@ func (d *Daemon) handleOIDCAuthenticationResult(r *http.Request, result *oidc.Au
 	return nil
 }
 
+// getSecrets gets a copy of the current, cluster-wide secrets.
+func (d *Daemon) getSecrets(ctx context.Context) (dbCluster.AuthSecrets, error) {
+	// Obtain a lock.
+	d.internalSecretsMu.Lock()
+	defer d.internalSecretsMu.Unlock()
+
+	// Get the expiry.
+	expiry, err := d.globalConfig.AuthSecretExpiry()
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if the current in-memory secrets are valid.
+	if d.internalSecrets.IsValid(expiry) {
+		// If valid, return a copy.
+		return slices.Clone(d.internalSecrets), nil
+	}
+
+	// Otherwise, start a transaction.
+	err = d.db.Cluster.Transaction(ctx, func(ctx context.Context, tx *db.ClusterTx) error {
+		// Get the secrets from the database.
+		q := `SELECT value, created_at FROM secrets WHERE entity_type = ? AND entity_id = 0 AND type = ? ORDER BY created_at DESC`
```

Comment:
Lets now have queries in the daemon itself and instead push this down into a cluster db function.

Reply:
Reduces cyclomatic complexity for one thing.

Reply:
Moved all the sql into `lxd/db/cluster/secrets.go`. My motivation behind keeping it all in one place was so that secret rotation can only happen in one place, but I appreciate it's harder to reason about.